core/src/main/java/org/apache/carbondata/core/datastore/page/UnsafeFixLengthColumnPage.java

```diff
@@ -326,9 +327,17 @@ public void encode(PrimitiveCodec codec) {
   }
 
   @Override
-  public byte[] compress(Compressor compressor) {
-    // TODO: use zero-copy raw compression
-    return super.compress(compressor);
+  public byte[] compress(Compressor compressor) throws MemoryException, IOException {
+    // use raw compression and copy to byte[]
+    int inputSize = pageSize << dataType.getSizeBits();
+    int compressedMaxSize = compressor.maxCompressedLength(inputSize);
+    MemoryBlock compressed = UnsafeMemoryManager.allocateMemoryWithRetry(compressedMaxSize);
```

Comment:
I think it is better to change the logic in UnsafeMemoryManager to simplify the usage code. In UnsafeMemoryManager, it will use unsafe memory based on
```
    boolean offHeap = Boolean.parseBoolean(CarbonProperties.getInstance()
        .getProperty(CarbonCommonConstants.ENABLE_OFFHEAP_SORT,
            CarbonCommonConstants.ENABLE_OFFHEAP_SORT_DEFAULT));
```
So I suggest to change this part to make one configuration only for unsafe. Then I will remove other unsafe configuration like ENABLE_UNSAFE_COLUMN_PAGE_LOADING
@ravipesala please check